libraries/chain/include/eosio/chain/contracts/contract_table_objects.hpp

```diff
@@ -148,7 +147,42 @@ namespace eosio { namespace chain { namespace contracts {
    typedef secondary_index<uint64_t,index_double_object_type,soft_double_less>::index_object  index_double_object;
    typedef secondary_index<uint64_t,index_double_object_type,soft_double_less>::index_index   index_double_index;
 
-} } }  // namespace eosio::chain::contracts
+} // ::contracts
+
+namespace config {
+   template<>
+   struct billable_size<contracts::key_value_object> {
+      static const uint64_t overhead = overhead_per_row_per_index_ram_bytes * 2;  ///< overhead for 2x indices internal-key and primary key
```

Comment:
We should probably also include the overhead of storing the `table_id_object`. This would overcharge for any table that isn't a singleton, but I don't know of any other simple solution to prevent abuse of memory by singleton tables.

Similar changes would be needed for the secondary index tables below.

Reply:
instead of including that overhead there (which became substantial) we are now billing the new table overhead to the first row's payer at the time it was created.  This does not track that rows payer as it changes.

When the last row of a table is deleted, we now reap the table and return the bytes to the original payer.